src/lib.rs

```diff
@@ -164,6 +168,10 @@ pub enum Error {
     #[cfg(feature = "alloc")]
     #[cfg_attr(docsrs, doc(cfg(feature = "alloc")))]
     OutOfMemory(TryReserveError),
+    /// An I/O error has occurred in [`write::strftime`].
+    #[cfg(feature = "std")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+    IoError(ErrorKind),
```

Comment:
For consistency with `TryReserveError` which also has a (unstable) `TryReserveErrorKind`, I think we should probably store the `io::Error` here. This would allow `Error::source` and downcasting to work.

Reply:
The problem is that `std::io::Error` does not derive `Clone, Eq, PartialEq`, so we need to remove these impl for the `strftime::Error` type.